x-pack/plugins/security_solution/common/endpoint/data_generators/base_data_generator.ts

```diff
@@ -33,6 +36,23 @@ export class BaseDataGenerator<GeneratedDoc extends {} = {}> {
     throw new Error('method not implemented!');
   }
 
+  /** Returns a future ISO date string */
+  protected randomFutureDate(from?: Date): string {
+    const now = from ? from.getTime() : Date.now();
+    return new Date(now + this.randomChoice(DAY_OFFSETS)).toISOString();
+  }
+
+  /** Returns a past ISO date string */
+  protected randomPastDate(from?: Date): string {
+    const now = from ? from.getTime() : Date.now();
+    return new Date(now - this.randomChoice(DAY_OFFSETS)).toISOString();
+  }
+
+  /** Generate either `true` or `false` */
+  protected randomBoolean(): boolean {
+    return this.randomChoice(BOOLEANS);
```

Comment:
I mean, ok. But why not `Math.random() < 0.5`?  This feels over-engineered

Reply:
we have these `random*` methods everywhere in the base generated, I just followed the same pattern.

Reply:
I mean, `randomBoolean()` is still a good fn to have.

Just.. have that function do `return Math.random() < 0.5` instead of grabbing from an array of enumerated boolean types (.. of course, there being two, but it seems really silly to exist).